src/systems/filecoin_vm/runtime/runtime.go

```diff
@@ -138,6 +139,16 @@ func (rt *VMContext) AbortAPI(msg string) Runtime_AbortAPI_FunRet {
 	return &Runtime_AbortAPI_FunRet_I{}
 }
 
+func (rt *VMContext) CreateActor_DeductGas() Runtime_CreateActor_DeductGas_FunRet {
+	if !rt._actorAddress.Equals(addr.InitActorAddr) {
+		rt.AbortAPI("Only InitActor may call rt.CreateActor_DeductGas")
+	}
```

Comment:
This does not seem like the right place for access control. Nothing bad happens anyway.

Reply:
The desired semantics are for only InitActor to be able to call this. Yes, it might be fine to allow this attack surface, but it might not -- the ability to trigger gas consumption interacts with lots of other potentially complicated logic. I don't think we should allow operations along these lines unless there's a good reason to.